File: check-server.mjs
import { Client } from 'ssh2';

const VPS = '66.42.113.157';
const USER = 'root';
const PASS = '@Ky8hTV{]NgqV@p)';
const SERVER_DIR = '/var/www/alshifa';

function run(conn, cmd) {
  return new Promise((resolve, reject) => {
    console.log(`[SSH] $ ${cmd}`);
    conn.exec(cmd, (err, stream) => {
      if (err) return reject(err);
      let out = '', err2 = '';
      stream.on('data', d => { out += d; process.stdout.write(d.toString()); });
      stream.stderr.on('data', d => { err2 += d; process.stderr.write(d.toString()); });
      stream.on('close', code => code === 0 ? resolve(out) : reject(new Error(`Exit ${code}: ${err2 || out}`)));
    });
  });
}

function put(conn, local, remote) {
  return new Promise((resolve, reject) => {
    conn.sftp((err, sftp) => {
      if (err) return reject(err);
      sftp.fastPut(local, remote, err2 => err2 ? reject(err2) : resolve());
    });
  });
}

const conn = new Client();
conn.on('ready', async () => {
  console.log('✅ SSH connected!\n');
  try {
    // Check PM2 config and server setup
    console.log('=== Check PM2 config ===');
    await run(conn, 'pm2 show alshifa 2>&1 | head -30');
    await run(conn, `ls -la ${SERVER_DIR}/.next/ 2>/dev/null | head -10`);
    await run(conn, `ls -la ${SERVER_DIR}/public/alshifa-debug.apk 2>/dev/null`);
    await run(conn, `ls ${SERVER_DIR}/node_modules/ 2>/dev/null | head -5`);
    await run(conn, `npm ls next --prefix ${SERVER_DIR} 2>&1 | head -5`);
  } catch (err) {
    console.error('Error:', err.message);
  } finally {
    conn.end();
  }
});
conn.on('error', err => { console.error('SSH Error:', err.message); process.exit(1); });
conn.connect({ host: VPS, port: 22, username: USER, password: PASS, readyTimeout: 15000 });

File: check_server.py
import paramiko
import time

HOST = "66.42.113.157"
USER = "root"
PASS = "@Ky8hTV{]NgqV@p)"

def main():
    ssh = paramiko.SSHClient()
    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
    ssh.connect(HOST, username=USER, password=PASS, timeout=30)

    # Check PM2 logs
    print("=== PM2 LOGS ===")
    stdin, stdout, stderr = ssh.exec_command("pm2 logs alshifa --lines 30 --nostream")
    print(stdout.read().decode())
    err = stderr.read().decode()
    if err:
        print(f"STDERR: {err}")

    # Wait a bit and check status
    time.sleep(3)
    print("\n=== PM2 STATUS ===")
    stdin, stdout, stderr = ssh.exec_command("pm2 status")
    print(stdout.read().decode())

    # Check website
    print("\n=== WEBSITE TEST ===")
    stdin, stdout, stderr = ssh.exec_command("curl -s -o /dev/null -w 'HTTP %{http_code}\n' http://localhost:3000/")
    print(stdout.read().decode().strip())

    ssh.close()

if __name__ == "__main__":
    main()
